lib/src/realm_property.dart

```diff
@@ -49,6 +49,9 @@ class SchemaProperty {
   /// Indicates that the property should be persisted under a different name
   final String? mapTo;
 
+  /// @nodoc
+  String get internalName => mapTo ?? name;
```

Comment:
That's introducing the problem which one should we use and when. Why not make `name` a getter which returns mapTo instead of name value?
```dart
String get name => mapTo ?? _name;
```

Reply:
or maybe we do need to differentiate in certain cases?

Reply:
We need the public name when setting up the Core schema.